improver/visibility/visibility_combine_cloud_base.py

```diff
@@ -0,0 +1,219 @@
+            to threshold and a cube of the probability of cloud base at ground level.
+        """
+        self.initial_scaling_value = initial_scaling_value
+        self.first_unscaled_threshold = first_unscaled_threshold
+
+    def separate_input_cubes(self, cubes: CubeList) -> tuple:
+        """Separate cubelist into a visibility cube and a cloud base at ground level cube.
+
+        Args:
+            cubes:
+                A cubelist only containing a cube of the probability of visibility
+                relative to thresholds and a cube of the probability of cloud base at
+                ground level
+        Returns:
+            A tuple containing a cube of the probability of visibility relative to threshold
+            and a cube of the probability of cloud base at ground level.
+        Raises:
+            ValueError:
+                If the input cubelist does not have exactly two cubes
+            ValueError:
+                If input cubelist doesn't contain a visibility cube and a cloud base at ground
+                level cube
+        """
+        visibility_cube = None
+        cloud_base_ground_cube = None
+
+        cubes = flatten(cubes)
+        cube_names = [cube.name() for cube in cubes]
+        if len(cubes) != 2:
+            raise ValueError(
+                f"""Exactly two cubes should be provided; one for visibility and one for
+                cloud base at ground level. Provided cubes are {cube_names}"""
+            )
+        else:
+            for cube in cubes:
+                if "visibility_in_air" in cube.name():
+                    visibility_cube = cube
+                elif "cloud_base" in cube.name():
+                    cloud_base_ground_cube = cube
+            if not visibility_cube or not cloud_base_ground_cube:
+                raise ValueError(
+                    f"""A visibility and cloud base at ground level cube must be provided.
+                    The provided cubes are {cube_names}"""
+                )
+
+        return (visibility_cube, cloud_base_ground_cube)
+
+    def get_scaling_factors(self, vis_thresholds: list) -> list:
+        """Calculates a scaling factor for every visibility threshold. The scaling factor
+        is determined differently depending on the threshold:
+
+        1) If the threshold is greater than or qual to first_unscaled_threshold then the
+        scaling factor is always 1.
+        2) If the threshold is less than first_unscaled_threshold then a scaling factor
+        is calculated by inputting the threshold into an inverted fourth level polynomial.
+        The constants in this curve have been defined such that a threshold equal to
+        first_unscaled_threshold gives a scaling factor of 1 and a threshold of 0m gives
+        a scaling factor equal to initial_scaling_value.
+
+        This distribution has been determined by experimentation and chosen for combining
```

Comment:
My internal debate when I was writing this plugin was how generic it should be. The specific distribution was chosen because it seemed to work with 4p5 oktas and that's the only cloud base we've experimented with scaling and the only one we're planning on scaling and combining with visibility. This distribution would work with other cloud bases but whether this is the best distribution for other cloud bases we haven't looked at e.g. it might be for 7.9 oktas you want something more linear.

Reply:
My internal debate when I was writing this plugin was how generic it should be. The specific distribution was chosen because it seemed to work with 4p5 oktas and that's the only cloud base we've experimented with scaling and the only one we're planning on scaling and combining with visibility. This distribution would work with other cloud bases but whether this is the best distribution for other cloud bases we haven't looked at e.g. it might be for 7.9 oktas you want something more linear.

Reply:
I'm not going to make this a blocker because this code works as intended right now, but I think I'll leave the comment open for the second reviewer in case they agree it should be 4p5 specific until further investigation.